module2.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+var myModule = (function () {
+// Private
+  var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday',
+  'Thursday', 'Friday', 'Saturday'];
+  var months = ['January', 'February', 'March', 'April', 'May',
+   'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+  var myDate = null;
+
+  function setDate (n) {
+    if (n === undefined) {
+      myDate = new Date();
+
+      return myDate;
+    } else if (n instanceof Date) {
```

Comment:
You don't want to store a reference to the date object passed in here, but rather you want to instantiate a new Date object using that:  `myDate = new Date(n)`  This prevents anything outside your module from modifying the date you're storing inside your module.